csrc/ir/base_nodes.cpp

```diff
@@ -218,7 +218,7 @@ PolymorphicValue Val::evaluate() {
 }
 
 bool Val::isZero() const {
-  return value().hasValue() && (bool)(value() == 0);
+  return value().hasValue() && (bool)(value() == 0.0);
```

Comment:
I don't know why this would work but an integer doesn't. I'm more than happy to open an issue to track this one cc'ing @zasdfgbnm